### ISSUE TITLE
Build fails failure to read artifact descriptor on clojure

### ISSUE THREAD

**Issue by awkay** [cljdoc/cljdoc#376]
When building the docs for Fulcro 3.1.5 the cljdoc build fails with:

https://repo.clojars.org/com/fulcrologic/fulcro/3.1.5-RC2/fulcro-3.1.5-RC2.pom {clojars {:url https://repo.clojars.org/}, central {:url http://central.maven.org/maven2/}}
Failed to read artifact descriptor for org.clojure:clojure:jar:1.10.0

See: https://circleci.com/gh/cljdoc/builder/16242

@martinklepsch looked into it some more and said:

> interesting, seems to happen with rebuilds too but locally everything works fine.
> ...
> I’ve looked into this some more but couldn’t find a good explanation yet.

**Comment by borkdude** [comment]
Has been happening with a lot of my recent projects as well. Example:

https://cljdoc.org/d/borkdude/edamame/0.0.10/doc/readme

https://circleci.com/gh/cljdoc/builder/16248

> Checking out: https://github.com/cljdoc/codox at e0cd26910704c416611fc81f43f890a26861c221
Failed to read artifact descriptor for org.clojure:tools.reader:jar:1.3.2 (edited)

**Comment by jimmyhmiller** [comment]
Same exact error on meander. Oldest build of meander that has the problem is from 8 days ago. https://circleci.com/gh/cljdoc/builder/15895

> Downloading remote jar...
https://repo.clojars.org/meander/epsilon/0.0.377/epsilon-0.0.377.pom {clojars {:url https://repo.clojars.org/}, central {:url http://central.maven.org/maven2/}}
Failed to read artifact descriptor for org.clojure:clojurescript:jar:1.10.439

**Comment by awkay** [comment]
I mean, if we take the message at face value it sounds like the build server is having trouble reaching that maven repo.

**Comment by jimmyhmiller** [comment]
Could it be because the maven repo isn't using https?  http://central.maven.org/maven2/ returns an error point to https://support.sonatype.com/hc/en-us/articles/360041287334

It is a recent change.

**Comment by awkay** [comment]
sounds quite probable

**Comment by martinklepsch** [comment]
Wohoo, @jimmyhmiller's PR fixed the issue. Thanks to everyone who helped troubleshoot this. 

@jimmyhmiller — I added you to the cljdoc community org on GitHub, you now have push access! :tada: Usually PRs are still a good idea but that means that you can contribute & merge PRs as you see fit 🙂

**Comment by georgiod9** [comment]
still having this error to this day

**Comment by lread** [comment]
@georgiod9 please provide a link to your failing build so I have something I can have a peek at.

**Comment by georgiod9** [comment]
> @georgiod9 please provide a link to your failing build so I have something I can have a peek at.

visit this https://github.com/georgiod9/clj-error-log/blob/master/README.md

**Comment by lread** [comment]
@georgiod9 I'm confused, how is this related to cljdoc?
Are you just sharing that you are hitting a similar problem for your own builds outside of cljdoc?

**Comment by lread** [comment]
But I can maybe help anyway... I see:
```
Downloading: org/clojure/clojure/1.4.0/clojure-1.4.0.pom from repository central at http://repo1.maven.org/maven2
```

I think that maven repos might need https these days.

I see:

```
 !     WARNING: No :min-lein-version found in project.clj; using 1.7.1.
         You probably don't want this!
-----> Installing Leiningen
       Downloading: leiningen-1.7.1-standalone.jar
       To use Leiningen 2.x, add this to project.clj: :min-lein-version "2.0.0"
       Writing: lein script
```
Try bumping leiningen to the current release, that might fix things for you.

**Comment by georgiod9** [comment]
> @georgiod9 I'm confused, how is this related to cljdoc? Are you just sharing that you are hitting a similar problem for your own builds outside of cljdoc?

The latter. I'm not famililar with clj at all. I'm handling a repo and trying to deploy on heroku.

**Comment by georgiod9** [comment]
> But I can maybe help anyway... I see:
> 
> ```
> Downloading: org/clojure/clojure/1.4.0/clojure-1.4.0.pom from repository central at http://repo1.maven.org/maven2
> ```
> 
> I think that maven repos might need https these days.
> 
> I see:
> 
> ```
>  !     WARNING: No :min-lein-version found in project.clj; using 1.7.1.
>          You probably don't want this!
> -----> Installing Leiningen
>        Downloading: leiningen-1.7.1-standalone.jar
>        To use Leiningen 2.x, add this to project.clj: :min-lein-version "2.0.0"
>        Writing: lein script
> ```
> 
> Try bumping leiningen to the current release, that might fix things for you.

ill try it out. thanks

**Comment by georgiod9** [comment]
Resolved by adding `:min-lein-version "2.0.0"` to my project.clj for heroku deployment

**Comment by lread** [comment]
Glad to hear you are sorted out @georgiod9.

When you run into a problem in the future, there are a bunch of ways to get general help from the [Clojure community](https://clojure.org/community/resources).
They really are awesomely friendly and welcoming.
My personal favourite is [Clojurians Slack](http://clojurians.net/).
